int32 serialID

int32 digDirection

int32 stop
